tmt/utils/__init__.py

```diff
@@ -4358,6 +4358,33 @@ def increment(
 
                 raise GeneralError(message) from error
 
+        # Handle GitHub-specific responses
+        # https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#exceeding-the-rate-limit
+        response = kwargs.get('response')
```

Comment:
is it possible to use `cast()` to introduce some types? See `error` above, it should be possible to do something like `response = cast(Optional[requests.Response], kwargs.get('response'))`. The same applies to `headers` below, `headers: dict[str, str]` should help linters & limit the amount of `Any` running around.

Reply:
And maybe `headers` we would get for free, `response.headers` should exist all the time, even if it'd be empty, i.e. no need for `getattr`?

Reply:
@happz Thanks, adding cast already found and issue :)  
I'm not sure about the headers, as that already has `CaseInsensitiveDict[str]` type.

Reply:
Yes, once linters become aware of the `response` type, I suppose they would get a better view of its attributes, and we wouldn't need to bother with `cast` calls.

Reply:
I would expect `kwargs.get('response')` returns `Any`, because of `**kwargs: Any`. As soon as `cast(requests.Response, ...)` is applied, mypy should be able to infer the rest, but without this hint, it'd all be just `Any`.